Sources/UtilityBeltNetworking/HTTPClient+Concurrency.swift

```diff
@@ -0,0 +1,91 @@
+// Copyright © 2023 SpotHero, Inc. All rights reserved.
+
+import Foundation
+
+/// A tuple returned from requests that return raw data.
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public typealias DataTaskSuccess = (request: URLRequest,
+                                    response: HTTPURLResponse,
+                                    data: Data)
+
+/// A tuple returned from requests that return decoded objects.
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public typealias DecodableTaskSuccess<T: Decodable> = (request: URLRequest,
+                                                       response: HTTPURLResponse,
+                                                       data: Data,
+                                                       value: T)
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+public extension HTTPClient {
+    // MARK: Data Response
+
+    /// Sends a request which fetches raw data from an endpoint.
+    /// - Parameters:
+    ///   - request: The `URLRequest` to make the request with.
+    ///   - validators: An array of validators that will be applied to the response.
+    ///   - interceptor: An object that can intercept the url request. Defaults to `nil`.
+    /// - Returns: The request used, the response and raw data.
+    func request(
```

Comment:
Will there be a way to cancel network requests?

Reply:
Ideally lol. Have a look at the last commit. Still need to test this though